src/ubx-message.toit

```diff
@@ -133,17 +419,30 @@ class Message:
 
   static is-valid-frame_ frame/ByteArray -> bool:
     // Check the sync bytes.
-    if frame[0] != 0xb5 or frame[1] != 0x62: throw INVALID-UBX-MESSAGE_
+    //if frame[0] != 0xb5 or frame[1] != 0x62: throw INVALID-UBX-MESSAGE_
+    // Don't throw here- let this function do as designed and allow
+    // the specific constructors etc determine what to do.  In every call of
+    // is-valid-frame_, it is thrown on false.
+    if frame[0] != 0xb5 or frame[1] != 0x62: return false
 
     // Check the payload length.
     length ::= LITTLE-ENDIAN.uint16 frame 4
     if not 0 <= length <= MAX-MESSAGE-SIZE_: return false
+    if frame.size != length + 8: return false
 
     ck-a ::= frame[frame.size - 2]
     ck-b ::= frame[frame.size - 1]
+
+    // AI suggested this close was bad:
+    //compute-checksum_ frame: | a b |
+    //  return ck-a == a and ck-b == b
+    //return false
+
+    // And suggested this replacement.  (Want to discuss)
+    ok := false
     compute-checksum_ frame: | a b |
-      return ck-a == a and ck-b == b
-    return false
+      ok = (ck-a == a) and (ck-b == b)
+    return ok
```

Comment:
AI's approach work, but old one was fine too.
I would replace the `return false` with `unreachable`.

```
compute-checksum_ frame: | a b |
  return ck-a == a and ck-b == b
unreachable
```
```